src/actions/cart-manager.ts

```diff
@@ -1,92 +1,60 @@
 import {retrieveSpecificProduct} from './DAL/product-db-actions';
 
-export const mainCart: { id: string, users: { userId: string, amount: number }[] }[] = [];
 
-export const addToCart = async (ip: string, productId: string, desiredAmount: number) => {
-    const product = await retrieveSpecificProduct(productId);
-    let availableAmount;
-    if (!product)
-        throw new Error('Item not found!');
-    let productInMainCart = findProductInMainCart(productId);
-    if (productInMainCart) {
-        availableAmount = product.quantity - getAmountOfItemInMainCart(productInMainCart);
-    } else {
-        availableAmount = product.quantity;
-        productInMainCart = {id: productId, users: []};
-        mainCart.push(productInMainCart);
+export const mainCart: Map<string, Map<string, number>> = new Map();
+
+export const addToCart = async (userId: string, productId: string, desiredQuantity: number) => {
+    const productInDB = await retrieveSpecificProduct(productId);
+    if (!productInDB) {
+        throw new Error('Product not found!');
+    }
+    if (!mainCart.has(productId)) {
+        mainCart.set(productId, new Map());
+    }
+    if (mainCart.get(productId).has(userId)) {
+        throw new Error('Product already in user cart. Please remove first to add new quantity');
```

Comment:
what if user want to change the amount in cart of this item?

Reply:
done